protobuf/v1/pool.proto

```diff
@@ -0,0 +1,58 @@
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+package mayastor.v1;
+
+// Create pool arguments.
+// Currently we support only concatenation of disks (RAID-0).
+message CreatePoolRequest {
+  string name = 1;           // name of the pool
+  repeated string disks = 2; // disk device paths or URIs to be claimed by the pool
+  PoolType pooltype = 3;     // type of the pool
+}
+
+// State of the storage pool (terminology comes from ZFS).
+enum PoolState {
+  POOL_UNKNOWN = 0;
+  POOL_ONLINE = 1;   // the pool is in normal working order
+  POOL_DEGRADED = 2; // the pool has experienced a failure but can still function
+  POOL_FAULTED = 3;  // the pool is completely inaccessible
+}
+
+enum PoolType {
+  Lvs = 0;  // pool backed by spdk lvs
+}
+
+// Storage pool properties
+message Pool {
+  string name = 1;            // name of the pool
+  repeated string disks = 2;  // absolute disk paths claimed by the pool
+  PoolState state = 3;        // current state of the pool
+  uint64 capacity = 5;        // size of the pool in bytes
+  uint64 used = 6;            // used bytes from the pool
+  PoolType pooltype = 7;      // type of the pool
+}
+
+// Destroy pool arguments.
+message DestroyPoolRequest {
+  string name = 1;  // name of the pool
+}
+
+// List of pools and their properties.
+message ListPoolsResponse {
+  repeated Pool pools = 1;  // list of the pools
+}
+
+message ListPoolOptions {
```

Comment:
@gila Can you please check the filter message? I guess this helps us implement things easily as the generated code has `Options`

```
pub struct ListPoolOptions {
    #[prost(oneof = "list_pool_options::Name", tags = "1, 2")]
    pub name: ::core::option::Option<list_pool_options::Name>,
    #[prost(oneof = "list_pool_options::Pooltype", tags = "3, 4")]
    pub pooltype: ::core::option::Option<list_pool_options::Pooltype>,
} 
```

Reply:
Yes either that, or use a hash table. Not sure what the keyword is for protobuf but i'm sure something like it exists.